### PR TITLE
Fixing query performance issue in Live Views

### LINKED ISSUE #30831: LIVE VIEWS are underusing available CPU
**Describe the situation**
What exactly works slower than expected?
Seems that LIVE VIEWS are underusing the available CPU.
 
**How to reproduce**
I'm using a GCP vm with 4 cpus, 32 GB ram and 727 GB disk.
Next code creates a table called 'test' and inserts random values into it. 
It will insert 1 billion rows as transactions between 21-09-01 and 21-09-10, with random values in amount column.
Then you have the selects to compare the performace of a live view that groups the amount transactioned each day, with the performance of a 'pure select' (same as the LIVE VIEW definition).
If you make an htop during the simple select you could see all cores working together, but at the first execution of select to the LIVE VIEW you will see cores with no use and a decrease on the performace of the query compared to the orginal select.
If you use more data or more complex querys with agg columns it turns worse.
 
```
-- create table test
CREATE TABLE test (
    id Int64,
    timestamp DateTime,
    amount Decimal(4,2)
) ENGINE = MergeTree()
ORDER BY timestamp AS
SELECT 
    *,
    toDateTime((modulo(rand(),10)/10)*(toUnixTimestamp('2021-09-10 00:00:00')-toUnixTimestamp('2021-09-01 00:00:00'))+toUnixTimestamp('2021-09-01 00:00:00')),
    toDecimal32(((modulo(rand(),100)/100)*(2)+(-1))*1000,2) as amount
FROM numbers(0, 1000000000)

-- just select and see htop
SELECT
    toStartOfDay(timestamp),
    MAX(amount) AS amount
FROM test AS mc
GROUP BY
    toStartOfDay(timestamp)
ORDER BY toStartOfDay(timestamp) DESC

-- create live view
CREATE LIVE VIEW test_lv WITH REFRESH 120
AS SELECT
    toStartOfDay(timestamp) as timestamp,
    MAX(amount) AS amount
FROM test AS mc
GROUP BY
    timestamp
ORDER BY timestamp DESC;

-- select and execute for the first time the live view (see htop)
SELECT * FROM test_lv ORDER BY timestamp DESC LIMIT 10;
```

**Which ClickHouse server version to use**
version 21.9.4 revision 54449

**Which interface to use, if matters**
clickhouse-client version 21.9.4.35 (official build)

**Expected performance**
I was expecting that live view executes over all the hardware available, with a performance comparable with the 'pure select' ones.

**Additional context**
I'm attaching some screenshots where you can check performance issue cleary

pure select performance
select-time

pure select cores
select-cores

live view performance
live-view-time

live view cores
live-view-cores

### PR DESCRIPTION
**Pull request by vzakaznikov**
Changelog category (leave one):
- Performance Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixing query performance issue in `LiveView` tables. Fixes #30831

Detailed description / Documentation draft:
Fixing performance issue reported in https://github.com/ClickHouse/ClickHouse/issues/30831 by
moving to `PullingAsyncPipelineExecutor`s instead of single threaded `PullingPipelineExecutor`s.

### CODE REVIEW

**Comment by vzakaznikov** [comment]
Tests show the following error
```
Code: 49. DB::Exception: Invalid number of columns in chunk pushed to OutputPort. Expected 3, found 1
Header: s Int32 Int32(size = 0), version Int32 Int32(size = 0), _version UInt64 UInt64(size = 0)
Chunk:  UInt64(size = 0)
```

File: src/Storages/LiveView/StorageLiveView.cpp [hunk 1]
```diff
@@ -385,7 +386,7 @@ bool StorageLiveView::getNewBlocks()
     auto builder = completeQuery(std::move(from));
     auto pipeline = QueryPipelineBuilder::getPipeline(std::move(builder));
 
-    PullingAsyncPipelineExecutor executor(pipeline);
+    PullingPipelineExecutor executor(pipeline);
```
Comment (vzakaznikov) [inline src/Storages/LiveView/StorageLiveView.cpp]: @KochetovNicolai, why would `PullingAsyncPipelineExecutor` handle blocks differently than `PullingPipelineExecutor`?

Comment (KochetovNicolai) [reply src/Storages/LiveView/StorageLiveView.cpp]: It shouldn't. Probably, there is a missing check in `PullingPipelineExecutor`. (Which is unusual cause the exception is from Port).

We need to fix header anyway.

Comment (vzakaznikov) [reply src/Storages/LiveView/StorageLiveView.cpp]: Not sure where the header needs to be fixed. Any ideas?

**Comment by vzakaznikov** [comment]
Hi @KochetovNicolai, I think this PR is ready to be merged!

**Comment by leandrogutierrez148** [comment]
Hi @vzakaznikov i've test the fix within the new release (https://github.com/ClickHouse/ClickHouse/releases/tag/v21.11.4.14-stable) and i'm still getting a worst performance VS a simple select, when selecting for the fist time a LIVE VIEW.

**Comment by vzakaznikov** [comment]
> Hi @vzakaznikov i've tested the fix within the new release (https://github.com/ClickHouse/ClickHouse/releases/tag/v21.11.4.14-stable) and i'm still getting a worse performance than a simple select, when selecting for the fist time a LIVE VIEW. Seems that only one core is active at the time the select is running.

Try testing using the build from this PR. For example, from https://clickhouse-builds.s3.yandex.net/31006/63c70f6281b4195c000eb2b9a35485dc06e26af2/clickhouse_build_check/report.html.

**Pull request merged**